packages/api/src/utils/logParser.ts

```diff
@@ -282,10 +277,30 @@ class VectorLogParser extends ParsingInterface<VectorLog> {
   }
 }
 
-class VectorMetricParser extends ParsingInterface<VectorMetric> {
+export const convertToStringMap = (obj: JSONBlob) => {
+  const mapped = mapObjectToKeyValuePairs(obj);
+  const converted_string_attrs: Record<string, string> = {};
+  for (let i = 0; i < mapped['string.names'].length; i++) {
+    converted_string_attrs[mapped['string.names'][i]] =
+      mapped['string.values'][i];
+  }
+  // at least nominally metrics should not have bool or number attributes, but we will
+  // handle and append them here just in case
+  for (let i = 0; i < mapped['number.names'].length; i++) {
+    converted_string_attrs[mapped['number.names'][i]] =
+      mapped['number.values'][i].toString();
+  }
+  for (let i = 0; i < mapped['bool.names'].length; i++) {
+    converted_string_attrs[mapped['bool.names'][i]] =
+      mapped['bool.values'][i].toString();
+  }
+  return converted_string_attrs;
+};
+
+class VectorMetricParser extends ParsingInterface<VectorMetric, MetricModel> {
   _parse(metric: VectorMetric): MetricModel {
     return {
-      _string_attributes: metric.b,
+      _string_attributes: convertToStringMap(metric.b),
```

Comment:
can we rollback this for now ? I'd prefer this to be applied in following PR. ideally, this PR should only update types

Reply:
Sure, I'll just put it back to `any`